File: qt/discuskeygen.cpp
#include "discuskeygen.h"

DiscusKeygen::DiscusKeygen(WifiNetwork * router ) : KeygenThread(router){}

void DiscusKeygen::run(){
    bool status;
    unsigned int routerSSIDint = router->getSSIDsubpart().toInt( &status , 16);
    if ( !status )
        return;
    QString result;
    result.setNum((routerSSIDint- 0xD0EC31)>>2);
    result = "YW0" + result;
    results.append(result);
}

File: src/org/exobel/routerkeygen/DiscusKeygen.java
package org.exobel.routerkeygen;

import android.content.res.Resources;
import android.os.Handler;

public class DiscusKeygen extends KeygenThread {

	public DiscusKeygen(Handler h, Resources res) {
		super(h, res);
	}

	static final int essidConst = 0xD0EC31;

	public void run(){
		int routerEssid = Integer.parseInt( router.getSSIDsubpart() , 16);
		int result  = ( routerEssid - essidConst )>>2;
		pwList.add("YW0" + Integer.toString(result));
		handler.sendEmptyMessage(RESULTS_READY);
		return;
		
	}

}
